src/deferred.js

```diff
@@ -46,6 +60,123 @@ jQuery.extend({
 						fns = null;
 					}).promise();
 				},
+				then: function( onFulfilled, onRejected, onProgress ) {
+					var maxDepth = 0;
+					function resolve( deferred, handler, depth ) {
+						return function() {
+							var that = this === promise ? undefined : this,
+								args = arguments,
+								fn = function() {
+									var returned, then;
+
+									// Support: Promises/A+ section 2.3.3.3.3
+									// https://promisesaplus.com/#point-59
+									// Ignore double-resolution attempts
+									if ( depth < maxDepth ) {
+										return;
+									}
+
+									try {
+										returned = handler.apply( that, args );
+
+										// Support: Promises/A+ section 2.3.1
+										// https://promisesaplus.com/#point-48
+										if ( returned === deferred.promise() ) {
+											throw new TypeError();
+										}
+
+										// Support: Promises/A+ sections 2.3.3.1, 3.5
+										// https://promisesaplus.com/#point-54
+										// https://promisesaplus.com/#point-75
+										// Retrieve `then` only once
+										then = returned &&
+
+											// Support: Promises/A+ section 2.3.4
+											// https://promisesaplus.com/#point-64
+											// Only check objects and functions for thenability
+											( typeof returned === "object" ||
+												typeof returned === "function" ) &&
+											returned.then;
+
+										if ( jQuery.isFunction( then ) ) {
+											maxDepth++;
+											then.call(
+												returned,
+												resolve( deferred, Identity, maxDepth ),
+												resolve( deferred, Thrower, maxDepth ),
+												deferred.notify
+											);
+										} else if ( Identity === handler ) {
+											deferred.resolveWith( that || deferred.promise(),
+												args );
+										} else {
+											deferred.resolve( returned );
+										}
+									} catch ( e ) {
+
+										// Support: Promises/A+ section 2.3.3.3.4.1
+										// https://promisesaplus.com/#point-61
+										// Ignore post-resolution exceptions
+										if ( depth + 1 >= maxDepth ) {
```

Comment:
This is the part in the story where it swallows errors. Some of the libs do complicated "next-turn" checking but I wonder if we could just do a `console.log` if the error that was thrown is one of the developer-screwup errors like `SyntaxError`. Something as simple as this?
```js
if ( /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error/.test(e) ) {
    console.log("[Caught exception] "+e);
    console.trace();
}
```
It could be in a public method that users could override perhaps?

Also note we only have one test case for `throw`n exceptions and none for browser-initiated exceptions.